XRTK-Core/Assets/XRTK.Tests/Core/TestFixture_03_ExtensionsTests.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) XRTK. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using NUnit.Framework;
+using UnityEngine;
+using XRTK.Extensions;
+
+namespace XRTK.Tests.Core
+{
+    public class TestFixture_03_ExtensionsTests
+    {
+        [Test]
+        public void Test_01_Array_AddItem_Extension()
```

Comment:
Can we also test to add the item at an random index and ensure that it's inserted in the correct postion?

Reply:
This is not an option it currently supports. Might add that in the future